src/cbc_sdk/platform/admin_grants.py

```diff
@@ -0,0 +1,471 @@
+
+    def _refresh(self):
+        """
+        Rereads the grant data from the server.
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        resp = self._cb.get_object(url)
+        self._info = resp
+        self._profiles = [Grant.Profile(self._cb, self, prof.get('profile_uuid', None), prof)
+                          for prof in resp.get('profiles', [])]
+        self._last_refresh_time = time.time()
+        return True
+
+    def _update_object(self):
+        """
+        Updates the grant data on the server.
+
+        Returns:
+            str: The principal URN of this grant object.
+        """
+        save_info = copy.deepcopy(self._info)
+        if 'principal' not in save_info:
+            raise ApiError("principal must be specified for a Grant object")
+        save_info['profiles'] = [copy.deepcopy(prof._info) for prof in self._profiles if 'profile_uuid' in prof._info]
+
+        if all([key in save_info for key in ['created_by', 'updated_by', 'create_time', 'update_time']]):
+            log.debug("Updating the Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject_single.format(self._cb.credentials.org_key, save_info['principal'])
+            ret = self._cb.put_object(url, save_info)
+        else:
+            log.debug("Creating a new Grant object for principal {0:s}".format(save_info['principal']))
+            url = self.urlobject.format(self._cb.credentials.org_key)
+            ret = self._cb.post_object(url, save_info)
+        update_return = self._refresh_if_needed(ret)
+
+        # add all profiles not yet part of grant
+        for profile in self._profiles:
+            if 'profile_uuid' not in profile._info:
+                profile._update_object()
+        return update_return
+
+    def _delete_object(self):
+        """Deletes the grant from the principal."""
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+        ret = self._cb.delete_object(url)
+        self._refresh_if_needed(ret)
+
+    @classmethod
+    def new_grant(cls, cb, principal):
+        """
+        Create a new grant object for the specified principal.
+
+        Args:
+            cb (CBCloudAPI): A reference to the CBCloudAPI object.
+            principal (str): The principal URN for the new grant object.
+
+        Returns:
+            Grant: The new grant object.
+        """
+        grant = Grant(cb, principal, {'principal': principal})
+        grant._full_init = True
+        return grant
+
+    @property
+    def profiles_(self):
+        """
+        Return the profiles associated with this grant.
+
+        Returns:
+            list: The profiles associated with this grant, each represented as a Profile object.
+        """
+        return self._profiles
+
+    def new_profile(self):
```

Comment:
Can we call the function `create_profile` and add an optional object where a user can specify an object for the profile instead of using the ProfileBuilder. This will let a user to have both as an option in case they already have a profile defined

Reply:
I can change the name, but why would they already have a profile defined?

Reply:
In an automation scenario they will already have their profile defined and just need to apply it to each grant they create for each new user. This will let them do that in one go instead of needing to call multiple functions in the ProfileBuilder

Reply:
This has been added.